docarray/array/array.py

```diff
@@ -99,6 +100,15 @@ def __len__(self):
     def __getitem__(self, item):
         if type(item) == slice:
             return self.__class__(self._data[item])
+        elif isinstance(item, Iterable):
+            offset_to_doc = self._get_offset_to_doc()
```

Comment:
this will be super slow right?

Reply:
We should it compute all the time or `cache` it

Reply:
the `_get_offset_to_doc()` you mean? we have to iterate over all the data at least once anyways; by doing this we iterate only once (during this call), if we were to not do this we would iterate once for every index given in the `Iterable`, so this is faster than the naive approach.

But yes, it could be made even faster through some caching